blockchain/src/history/history_store.rs

```diff
@@ -648,13 +655,38 @@ impl HistoryInterface for HistoryStore {
         let tree = MerkleMountainRange::new(MMRStore::with_read_transaction(
             &self.hist_tree_table,
             txn,
-            epoch_number,
+            Policy::epoch_at(block_number),
         ));
 
         // Return the history root.
         tree.get_root().ok()
     }
 
+    fn tx_in_validity_window(
+        &self,
+        tx_hash: &Blake2bHash,
+        validity_window_start: u32,
+        txn_opt: Option<&TransactionProxy>,
+    ) -> bool {
+        // Get a vector with all transactions corresponding to the given hash.
+        let ext_hash_vec = self.get_hist_tx_by_hash(tx_hash, txn_opt);
+
+        // If the vector is empty then we have never seen a transaction with this hash.
+        if ext_hash_vec.is_empty() {
+            return false;
+        }
+
+        for hist_tx in ext_hash_vec {
+            // If the transaction is inside the validity window, return true.
+            if hist_tx.block_number >= validity_window_start {
```

Comment:
Currently, we only check whether the transaction occurred *after* the validity window started.
However, it could potentially still fall outside the window if it occurred after it finished.
The second boundary check seems missing.

Reply:
Here I'm not doing anything different than what we were doing before, i.e: I just moved this code: https://github.com/nimiq/core-rs-albatross/blob/albatross/blockchain/src/blockchain/wrappers.rs#L196
to the history store file.

Do you suggest that we used to have a bug? 
What do you mean by "if it occurred after it finished." ?

Reply:
I don't know what that constant is, but that would be the check. This does not make any assumptions on where the chain is so the validity window that starts at `validity_window_start` could be a long time ago. Thus there are transactions after this window which would be valid.

i.e.
```suggestion
            if hist_tx.block_number >= validity_window_start && hist_tx.block_number <= validity_window_start + VALIDITY_WINDOW {
```

Depending on how it was used this would potentially not be buggy, but checking both boundaries is certainly more sound.